### DISCUSSION TITLE
Custom data visualization

### DISCUSSION THREAD

**Discussion by pkp124** [allure-framework/allure2#2307] [General Discussion]
Hello,

Is it possible to visualize some custom data across runs, for ex performance numbers for some benchmark tests, using allure. Assuming this data is part of the output files in some format.

Thanks

**Comment by baev** [comment]
You can use custom HTML attachments if you want to add data to particular tests. There is an option to create a report plugin to create a custom tab/widget/graph in the report, but that would require managing a custom Allure Commandline distribution.

BTW, suppose you're looking for ways to display load testing results in Allure (such as JMeter). In that case, we have an extension for Allure TestOps users that would process JMeter results, aggregate them, and display them in the same report format.

**Reply by pkp124** [reply]
I was thinking of something simpler, like support for openmetrics data format. I suppose that requires a plugin to be implemented or for data to be converted to JMeter format.
